backend/src/routes/api/k8s/pass-through.ts

```diff
@@ -16,7 +16,42 @@ type PassThroughData = {
 export const isK8sStatus = (data: unknown): data is K8sStatus =>
   (data as K8sStatus).kind === 'Status';
 
-export const passThrough = <T extends K8sResourceCommon>(
+const passThroughCatch = (fastify: KubeFastifyInstance) => (error: Error) => {
+  let errorMessage = 'Unknown error';
+  if (error instanceof ProxyError) {
+    errorMessage = error.message || errorMessage;
+    switch (error.proxyErrorType) {
+      case ProxyErrorType.CALL_FAILURE:
+        fastify.log.error(`Kube parsing response error: ${errorMessage}`);
+        throw { code: 500, response: error };
+      case ProxyErrorType.HTTP_FAILURE:
+        fastify.log.error(`Kube request error: ${errorMessage}`);
+        throw { code: 500, response: error };
+      default:
+      // unhandled type, fall-through
+    }
+  } else if (!(error instanceof Error)) {
+    errorMessage = JSON.stringify(error);
+  }
+
+  fastify.log.error(`Unhandled error during Kube call: ${errorMessage}`);
+  throw error;
+};
+
+export const passThroughText = (
+  fastify: KubeFastifyInstance,
+  request: OauthFastifyRequest,
+  data: PassThroughData,
+): Promise<string> => {
+  return (
+    proxyCall(fastify, request, data)
+      // TODO: is there bad text states that we want to error out on?
+      .then((rawData) => rawData) // noop intentionally until above inquiry is figured out
```

Comment:
I'm not going to change this logic atm -- we can see what happens when @manaswinidas does more of the logging work.